packages/x-data-grid-premium/src/hooks/features/chartsIntegration/useGridChartsIntegration.tsx

```diff
@@ -195,17 +193,20 @@ export const useGridChartsIntegration = (
       }
 
       const columns = gridColumnLookupSelector(apiRef);
-      const unwrappedColumnGroupingModel = gridColumnGroupsUnwrappedModelSelector(apiRef);
+      const columnGroupPath = apiRef.current.getColumnGroupPath(field);
+      const columnGroupLookup = apiRef.current.getAllGroupDetails();
 
       const column = columns[field];
 
       const columnName = column?.headerName || field;
-      if (!pivotActive || !unwrappedColumnGroupingModel[field]) {
+      if (!pivotActive || !columnGroupPath) {
         return columnName;
       }
 
-      const groupPath = unwrappedColumnGroupingModel[field].slice(-1)[0];
-      return [columnName, ...groupPath.split(COLUMN_GROUP_ID_SEPARATOR)].join(' - ');
+      const groupNames = columnGroupPath.map(
+        (group) => columnGroupLookup[group].headerName || group,
```

Comment:
If the `headerName` doesn't exist for some reason, it may end up showing the separator as part of the label, how about still using `COLUMN_GROUP_ID_SEPARATOR` to split `group` parameter?

<img width="249" height="102" alt="image" src="https://github.com/user-attachments/assets/5ea057a4-1cf7-4db4-bc99-50818f425358" />

Or is `headerName` always defined? In which case, we should probably add a non-null assertion here rather than the `group` fallback.